middleware/token.middleware.js

```diff
@@ -0,0 +1,37 @@
+import { checkCookie } from "../utils/checkCookie.js";
+import {
+  isAccessValidHandle,
+  isRefreshValidHandle,
+} from "../utils/tokenCheck.js";
+import { refreshToken, token } from "../utils/tokenGenerator.js";
+import {
+  accessTokenOptions,
+  refreshTokenOptions,
+} from "../constants/constants.js";
+
+export const checkToken = async (req, res, next) => {
+  try {
+    const [accessTokenFromCookie, refreshTokenFromCookie] = checkCookie(
+      req.headers["cookie"],
+    );
+    if (!accessTokenFromCookie && !refreshTokenFromCookie) {
+      res.status(401).send();
+    }
+    const isAccessValid = await isAccessValidHandle(accessTokenFromCookie);
+    if (!isAccessValid) {
```

Comment:
Also your logic looks too complicated. 
<img width="778" alt="Screen Shot 2021-12-14 at 17 11 05" src="https://user-images.githubusercontent.com/47276289/146014439-135b7d3c-0f4b-4920-a180-37b085a02f92.png">
This variant is much simplier